src/big/big_decimal.cr

```diff
@@ -204,6 +204,7 @@ struct BigDecimal < Number
   # ```
   def div(other : BigDecimal, max_div_iterations = DEFAULT_MAX_DIV_ITERATIONS) : BigDecimal
     check_division_by_zero other
+    other.factor_powers_of_ten
```

Comment:
Since calling `factor_powers_of_ten` is changing internal object state, should we `dup` the *other* before calling it?